### ISSUE TITLE
Support platform versions back to macOS 10.15,  iOS 13.0, tvOS 13.0, and watchOS 6.0

### ISSUE THREAD

**Issue by czechboy0** [apple/swift-openapi-generator#62]
Let's see how much work it'd be to reduce the deployment targets, macOS 13->11, and so on.

This would be all of the repos - generator, runtime, and transports.

// Edit: Possibly we could go as low as `.macOS(.v10_15), .iOS(.v13), .tvOS(.v13), .watchOS(.v6)`.

**Comment by andrewse02** [comment]
Hi, I actually did this today! You may have already seen that. It was very simple. There was one instance I found of iOS 16+/macOS 13+ code being used in `swift-openapi-runtime`. Here's my [commit](https://github.com/andrewse02/swift-openapi-runtime-iOS14/commit/b38b854e3b80787c507caef273924cee9e7e8ef0). It was this right here in `Converter+Client.swift`:
```swift
for parameter in parameters {
    renderedString.replace(
        "{}",
        with: parameter.description,
        maxReplacements: 1
    )
}
```
It was simple enough to wrap this in a `if #available`. Would you like me to make a PR from my fork?

**Comment by simonjbeaumont** [comment]
I think we should be OK to relax the deployment version.

@andrewse02 thanks for looking into it and for offering to put a PR together—that'd be great. 🙌

**Comment by denil-ct** [comment]
I was thinking of reducing it to iOS 12 as one of our apps still had 12 as the deployment target, and it was easy to change it for the generator and runtime. But for the transport since there is no concurrency support, I've hit a roadblock :( 
Till iOS 13, there aren't any major issues, so I think we can reduce it to even iOS 13.

If I need to use it in a iOS 12 environment is it doable in some other way? Maybe with the new macros can we make it generate both the completion style and async/await styles for the `client.send` method?

**Comment by czechboy0** [comment]
@andrewse02 Since @denil-ct was able to reduce the URLSession transport's deployment target by 3 versions in https://github.com/apple/swift-openapi-urlsession/pull/7, can you try to do the same for the runtime and generator as well? It's okay if transports are _more_ restrictive, as they're usually more tied to specific platform features, but if it's _less_ restrictive, then adopters won't be able to use it anyway (due to the runtime package requiring a higher deployment target).

So let's try to go for the following: `.macOS(.v10_15), .iOS(.v13), .tvOS(.v13), .watchOS(.v6)`.

@denil-ct If you'd like, you could try a similar PR in https://github.com/swift-server/swift-openapi-async-http-client

**Comment by denil-ct** [comment]
Sure @czechboy0, will check it out.

**Comment by lucas-quinn** [comment]
I tried this yesterday, although the documentation isn't that good. However, after reading the source code, I was able to make it work fine. Great job! There's just one thing preventing our company from adopting it: if possible, please lower it to at least iOS 15.0, which is the highest version many devices can get.

Additionally, iOS 15.0 (90.9% adoption) would be much safer than 16.0 (74.9% adoption) in terms of current user adoption. But if there is no difficulty in lowering it to iOS 13.0, that would be even better.

**Comment by simonjbeaumont** [comment]
@lucas-quinn wrote:
> Great job! There's just one thing preventing our company from adopting it: if possible, please lower it to at least iOS 15.0, which is the highest version many devices can get.
> 
> Additionally, iOS 15.0 (90.9% adoption) would be much safer than 16.0 (74.9% adoption) in terms of current user adoption. But if there is no difficulty in lowering it to iOS 13.0, that would be even better.

Thanks for adding your datapoint here. It looks like the current plan (in flight) is to drop to minimum of iOS 13 so that should suit you :)

> I tried this yesterday, although the documentation isn't that good. However, after reading the source code, I was able to make it work fine. 

We'd love to understand how we could improve the documentation. Could you elaborate on what was missing and forced you to inspect the code? 🙏

**Comment by czechboy0** [comment]
cc @adam-fowler @0xTim for the Hummingbird and Vapor transports, in case you also want to reduce the deployment target (once https://github.com/apple/swift-openapi-runtime/pull/18 lands and is tagged).

**Comment by czechboy0** [comment]
Reopening until the generator and URLSession/AHC transports are updated as well.

**Comment by denil-ct** [comment]
I guess we can close this? @czechboy0

**Comment by czechboy0** [comment]
Runtime, URLSession, and AHC have been released, but not this repo yet: #63 - let's wait for that before we close this issue.
